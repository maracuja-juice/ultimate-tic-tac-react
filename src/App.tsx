import * as React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Game from './routes/Game/Game';
import AnalysisGameRoute from './routes/AnalysisGame/AnalysisGame';
import AnalysisOverview from './routes/AnalysisOverview/AnalysisOverview';
import routes from './routes/routes';
import { Navigation } from './components/Navigation/Navigation';
import HowToPlay from './routes/HowToPlay/HowToPlay';

export const App = () => {
  return (
    <BrowserRouter>
      <Navigation />
      <main>
        <Switch>
          <Route path={routes.AnalysisLatest} component={AnalysisGameRoute} />
          <Route path={routes.AnalysisParam} component={AnalysisGameRoute} />
          <Route path={routes.AnalysisOverview} component={AnalysisOverview} />
          <Route path={routes.Home} component={Game} />
        </Switch>
        <Route path={routes.HowToPlay} component={HowToPlay} />
      </main>
    </BrowserRouter>
  );
};
