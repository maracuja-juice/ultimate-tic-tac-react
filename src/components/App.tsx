import * as React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import GithubCorner from 'react-github-corner';
import { Game } from '../views/Game';
import Analysis from '../views/Analysis';
// import individual material component styles here.
import '@material/button/dist/mdc.button.min.css';

export class App extends React.Component {

    lightBlue = window.getComputedStyle(document.body).getPropertyValue('--tropical-teal').trim();

    render() {
        return (
            <>
                <GithubCorner
                    href="https://github.com/ultimate-ttt/ultimate-ttt"
                    bannerColor={this.lightBlue}
                    octoColor="#fff"
                    size={80}
                    direction="right"
                />
                <Router>
                    <div>
                        <Route exact={true} path="/" component={Game}/>
                        <Route exact={true} path="/analysis/:gameId" component={Analysis} />
                    </div>
                </Router>
            </>

        ); /* TODO two different components for analytics? */
    }
}