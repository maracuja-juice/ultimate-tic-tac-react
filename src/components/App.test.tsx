import * as React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  it('should not explode', () => {
    const app = shallow(<App />);
    expect(app).not.toBeNull();
  });
});
