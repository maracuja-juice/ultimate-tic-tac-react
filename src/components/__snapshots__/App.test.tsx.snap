// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AnalysisGame)",
          "type": [Function],
        }
      }
      exact={true}
      path="/analysis/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Game)",
          "type": [Function],
        }
      }
      path="/"
    />
  </Switch>
</BrowserRouter>
`;
